functions/uploads.py

```diff
@@ -76,30 +88,25 @@ def ingest_upload(event: dict, context: BackgroundContext):
                     f"Invalid assay metadata: missing protocol identifier ({prism.PROTOCOL_ID_FIELD_NAME})."
                 )
 
-        def do_copy(urls):
-            upload_url, target_url = urls
-            with saved_failure_status(job, session):
-                # Copy the uploaded GCS object to the data bucket
-                destination_object = _gcs_copy(
-                    GOOGLE_UPLOAD_BUCKET, upload_url, GOOGLE_DATA_BUCKET, target_url
-                )
-            return destination_object
+        print(
+            f"Detected completed upload job (job_id={job_id}) for user {job.uploader_email}"
+        )
 
-        uuids = []
-        url_mapping = []
-        for upload_url, target_url, uuid in job.upload_uris_with_data_uris_with_uuids():
-            url_mapping.append((upload_url, target_url))
-            uuids.append(uuid)
+        url_bundles = [
+            URLBundle(*bundle) for bundle in job.upload_uris_with_data_uris_with_uuids()
+        ]
```

Comment:
This is a refactor that shouldn't change the behavior of the function.

Reply:
nit: consider making `upload_uris_with_data_uris_with_uuids` returning `URLBundle`s